tests/test_bgp_neighbor_af.rb

```diff
@@ -552,9 +697,95 @@ def send_community(af, dbg)
 
     v = 'both'
     af.send_community = v
+    assert_equal('both', af.send_community,
+                 "Test 9. #{dbg} Failed to set '#{v}' from None")
+
+    v = af.default_send_community
+    af.send_community = af.default_send_community
+    assert_equal(v, af.send_community,
+                 "Test 10. #{dbg} Failed to set state to default")
+  end
+
+  def send_comm_xr(af, dbg)
+    v = 'both'
+    af.send_community = v
+    assert_equal(v, af.send_community,
+                 "Test 1a. #{dbg} Failed to set '#{v}' from None")
+    af.send_community = v
+    assert_equal(v, af.send_community,
+                 "Test 1b. #{dbg} Failed to set '#{v}' from None")
+    v = 'extended'
+    af.send_community = v
+    assert_equal('send-extended-community-ebgp', af.send_community,
```

Comment:
Yes with your suggestions I was able to dump the XR specific testing and use the nexus test code.